homeassistant/components/proliphix/climate.py

```diff
@@ -100,7 +102,7 @@ def target_temperature(self):
     @property
     def hvac_mode(self):
         """Return the current state of the thermostat."""
-        state = self._pdp.hvac_mode
+        state = self._pdp.hvac_state
```

Comment:
From how I read the [prolifix api docs](https://github.com/snappa/thermostat.rb/blob/master/docs/PDP_API_R1_11.pdf) the `thermHvacState` with OID 4.1.2, here represented by `self._pdp.hvac_state`, represents the state of the HVAC, ie if it's inactive, cooling or heating at the current time. In home assistant this should be return by the `hvac_state` property.

The `hvac_mode` property, here, should return what mode the HVAC is set to, ie if the user wants it to heat or cool etc.

https://github.com/sdague/proliphix/blob/da2cb3f2193ecc5887ec0eac7265e681a59f6186/proliphix/proliphix.py#L36-L53

We want to use `thermHvacMode`, OID 4.1.1, for `hvac_mode`. The only public interface that uses that in the library are these two properties, `is_cooling` and `is_heating`:
https://github.com/sdague/proliphix/blob/da2cb3f2193ecc5887ec0eac7265e681a59f6186/proliphix/proliphix.py#L134-L140

It would be good if the library could expose all the modes, since there seems to be off and auto besides heating and cooling.

Reply:
Not sure how to test the various methods or actions available but these changes now allow my system to read the proliphix thermostat status.

Reply:
Hi Martin,

I think what you want requires a change to the proliphix package from pypi.org?  I'm willing to look into that, but it seems to go a bit beyond the original bug fix to get the basic thermostat functionality running on home assistant.

Regards,

Martijn

Reply:
For this PR it's enough to add a property `hvac_action` that returns `self._pdp.hvac_state`, and check `self._pdp.is_cooling` and self._pdp.is_heating` here in `hvac_mode`.

Reply:
Ok, I changed the code. This seems to work in my setup.